docs/src/main/asciidoc/kafka-streams-guide.adoc

```diff
@@ -0,0 +1,1141 @@
+
+=== Topic Configuration
+
+The two streams are mapped to Kafka Apache topics using the Quarkus configuration file `application.properties`.
+For that, add the following to the file `producer/src/main/resources/application.properties`:
+
+[source]
+----
+kafka.bootstrap.servers=localhost:9092
+
+mp.messaging.outgoing.temperature-values.connector=smallrye-kafka
+mp.messaging.outgoing.temperature-values.key.serializer=org.apache.kafka.common.serialization.IntegerSerializer
+mp.messaging.outgoing.temperature-values.value.serializer=org.apache.kafka.common.serialization.StringSerializer
+
+mp.messaging.outgoing.weather-stations.connector=smallrye-kafka
+mp.messaging.outgoing.weather-stations.key.serializer=org.apache.kafka.common.serialization.IntegerSerializer
+mp.messaging.outgoing.weather-stations.value.serializer=org.apache.kafka.common.serialization.StringSerializer
+----
+
+This configures the Kafka bootstrap server, the two topics and the corresponding (de-)serializers.
+More details about the different configuration options are available on the https://kafka.apache.org/documentation/#producerconfigs[Producer configuration] and https://kafka.apache.org/documentation/#consumerconfigs[Consumer configuration] section from the Kafka documentation.
+
+== Creating the Aggregator Maven Project
+
+With the producer application in place, it's time to implement the actual aggregator application,
+which will run the Kafka Streams pipeline.
+Create another project like so:
+
+[source, subs=attributes+]
+----
+mkdir aggregator && cd aggregator && \
+mvn io.quarkus:quarkus-maven-plugin:{quarkus-version}:create \
+    -DprojectGroupId=org.acme \
+    -DprojectArtifactId=kafka-streams-quickstart-aggregator \
+    -Dextensions="kafka-streams,resteasy-jsonb" \
+    && cd ..
+----
+
+This creates the `aggregator, project with the Quarkus extension for Kafka Streams and with RESTEasy support for JSON-B.
+
+=== The Pipeline Implementation
+
+Let's begin the implementation of the stream processing application by creating
+a few value objects for representing temperature measurements, weather stations and for keeping track of aggregated values.
+
+First, create the file `aggregator/src/main/java/org/acme/quarkus/sample/kafkastreams/model/WeatherStation.java`,
+representing a weather station, with the following content:
+
+[source, java]
+----
+package org.acme.quarkus.sample.kafkastreams.model;
+
+import io.quarkus.runtime.annotations.RegisterForReflection;
+
+@RegisterForReflection
+public class WeatherStation {
+
+    public int id;
+    public String name;
+}
+----
+
+Then the file `aggregator/src/main/java/org/acme/quarkus/sample/kafkastreams/model/TemperatureMeasurement.java`,
+representing temperature measurements for a given station:
+
+[source, java]
+----
+package org.acme.quarkus.sample.kafkastreams.model;
+
+import java.time.Instant;
+
+public class TemperatureMeasurement {
```

Comment:
Don't you need the `@RegisterForReflection`

Reply:
No, it's only instantiated once in `KafkaStreamsPipeline`, calling the constructor.